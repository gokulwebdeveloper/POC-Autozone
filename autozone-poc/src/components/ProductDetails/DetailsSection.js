import React, {useEffect} from 'react';
import '../../matchMedia.mock';
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import '../ProductDetails/ProductDetails.css';
import Button from '../../utlis/button/Button';
import PropTypes from 'prop-types';
import Rating from 'react-rating';
import { connect } from 'react-redux';
import {addToCart} from '../../Redux/actions';
import moment from 'moment';
import add from "../../assets/img/core-img/add.svg"
import minus from "../../assets/img/core-img/minus.svg"

const DetailsSection = ({currentProduct, dispatchAddToCart}) => {
    const nextDay = new Date();
    const today = new Date();
    nextDay.setDate(nextDay.getDate() + 1);

    const deliveryMessage = "Order Before 9:00 PM "+moment(today).format('MMM DD')+" for "+moment(nextDay).format('MMM DD')+" delivery";

    useEffect( ()=>{
        
        updateFavoriteIco(currentProduct);
        let txtcartquantity = document.getElementById("txtcartquantity");

        if(txtcartquantity)
        {
            if(currentProduct.product_instock)
                currentProduct.quantity ? txtcartquantity.value = currentProduct.quantity : txtcartquantity.value = 1;
            else
                txtcartquantity.value = 0;
        }

        let outOfStock = document.getElementById("outOfStock");
        if(outOfStock)
        {
            if(currentProduct.product_instock)
                outOfStock.style.visibility = "hidden";
            else
                outOfStock.style.visibility = "visible";
        }
     });

    return (
        <section className="single_product_details_area d-flex align-items-center">
            <div className="single_product_thumb clearfix">
            <Slider 
                    dots={true}
                    infinite={true}
                    speed={1000}
                    slidesToShow={1}
                    slidesToScroll={1}
                    swipeToSlide={true}
                    initialSlide={0}
                    autoplay={true}
                    autoplaySpeed={4000}
                    arrows={true}
                    >

                    {
                    currentProduct.product_image ? 

                    currentProduct.product_image.map((image, index) => {
                    return (
                        <img src={image} alt={currentProduct.product_name} key={index}/>
                        )
                    })
                    
                    : "No Images"
                    }
                </Slider>
            </div>

            {/* <!-- Single Product Description --> */}
            <div className="single_product_desc clearfix">
                <span>{currentProduct.product_brand}</span>
                    <h2>{currentProduct.product_name}</h2>
                    <p className="rating-row">
                         <Rating
                    initialRating={currentProduct.rating}
                    readonly
                    className="rating"
                    emptySymbol="fa fa-star-o fa-2x" 
                    fullSymbol="fa fa-star fa-2x"
                    fractions={2}
                    />
                    <span className="rating-number">({currentProduct.rating})</span>
                    <span className="warranty">{currentProduct.warranty}</span>
                    </p>

                    <h6 id="outOfStock" className="outofstock">Out of stock</h6>
                   
                <p className="product-desc">{currentProduct.product_desc}</p>
            
                    <p className="product-price"><span className="old-price">${currentProduct.original_price}</span>
                    <span className="new-price"> $</span>
                        <span id="cartprice" className="new-price">{currentProduct.offer_price}</span></p>
                
                <div id="divQty" className="d-flex">
                <div className="d-flex">
                    <button id="btnDecreaseQty" className="quantitybutton"
                            onClick={()=>onDecreaseQty(currentProduct)} 
                            disabled={currentProduct.product_instock === true ? false : true}>
                        <img src={minus} alt="decrease button"/>
                    </button>
                </div>
                <div className="d-flex">
                    <input id="txtcartquantity" className="form-control quantitybox"
                    onChange={()=>onQtyChange(currentProduct)} onBlur={()=>onQtyLostFocus(currentProduct)} maxLength="1"
                    disabled={currentProduct.product_instock === true ? false : true}/>
                </div>
                <div className="d-flex">
                    <button className="quantitybutton"
                    onClick={()=>onIncreaseQty(currentProduct)}
                    disabled={currentProduct.product_instock === true ? false : true}>
                        <img src={add} alt="increase button"/>
                    </button>
                </div>
                </div>
                <div id="pricePerItem" className="priceperitem" ></div>
                
            <div className="orderByMessage">{deliveryMessage}</div>
                    {/* <!-- Cart & Favourite Box --> */}
                    <div className="cart-fav-box d-flex align-items-center">
                        <Button id="btnAddtoCart" btnText="Add to Cart" 
                        disabled={currentProduct.product_instock === true ? false : true}
                        onClick={()=>addToCartFn(currentProduct, dispatchAddToCart)}/>
                        {/* <!-- Favourite --> */}
                        <div className="product-favourite ml-4" onClick={changeFavStatus}>
                            <a id="favme" className="favme fa fa-heart"></a>
                        </div>
                    </div>
                    
            </div>
        </section>
    );
};

const maxAllowedQty = 5;

function addToCartFn(currentProduct, dispatchAddToCart)
{
    let txtcartquantity = document.getElementById("txtcartquantity");

    let qty = parseInt(txtcartquantity.value);

    addKeyValue(currentProduct,"quantity", qty);
    addKeyValue(currentProduct,"price", currentProduct.offer_price);

    dispatchAddToCart(currentProduct);
}

function updateFavoriteIco(currentProduct){
let currentCntrl = document.getElementById("favme");

if(currentCntrl!=null)
{
    if(currentProduct.isFavourite)
        currentCntrl.className = "favme fa fa-heart active";
    else
        currentCntrl.className = "favme fa fa-heart";
}
}

function changeFavStatus() {
let currentCntrl = document.getElementById("favme");

if(currentCntrl==null)
    return;

if(!currentCntrl.className.includes("active"))
    currentCntrl.className = "favme fa fa-heart active";
else
    currentCntrl.className = "favme fa fa-heart";
}

function onQtyChange(currentProduct)
{
let currentCntrl = document.getElementById("txtcartquantity");

if(currentCntrl==null)
    return;

let qty = parseInt(currentCntrl.value);

if(currentCntrl.value == null)
{
    currentCntrl.value = 1;
    updateEach(qty, currentProduct);
}
else if(qty > maxAllowedQty)
{
    currentCntrl.value = 1;
    updateEach(qty, currentProduct);
}
else if(qty > 1 && qty <= maxAllowedQty)
{
    updateCartPrice(currentProduct);
    updateEach(qty, currentProduct);
}
else
    return;
}

function onQtyLostFocus(currentProduct)
{
let currentCntrl = document.getElementById("txtcartquantity");

if(currentCntrl==null)
    return;

if(currentCntrl.value == null || currentCntrl.value == "")
{
    currentCntrl.value = 1;
    updateCartPrice(currentProduct);  
}
}

function onIncreaseQty(currentProduct)
{
let currentCntrl = document.getElementById("txtcartquantity");

if(currentCntrl==null)
    return;

if(currentCntrl.value == null || currentCntrl.value == "")
    currentCntrl.value = 1;
else
{
    let val = parseInt(currentCntrl.value);
    
    if(val < maxAllowedQty)
    {
        currentCntrl.value = val + 1;
        updateCartPrice(currentProduct);
    }
    else
        return;
}
}

function onDecreaseQty(currentProduct)
{
let currentCntrl = document.getElementById("txtcartquantity");

if(currentCntrl==null)
    return;

if(currentCntrl.value == null || currentCntrl.value == "")
    currentCntrl.value = 1;
else
{
    let val = currentCntrl.value;
    if(val > 1)
    {
        currentCntrl.value = val - 1;
        updateCartPrice(currentProduct);
    }
    else
        return;
}
}

function updateCartPrice(currentProduct)
{
let currentCntrl = document.getElementById("txtcartquantity");

if(currentCntrl==null)
    return;

let cartCntrl = document.getElementById("cartprice");

if(cartCntrl==null)
    return;

let qty = parseInt(currentCntrl.value);
let price = parseFloat(currentProduct.offer_price);
let total = qty * price;
cartCntrl.innerHTML = total;
updateEach(qty, currentProduct);
}

function updateEach(qty, currentProduct)
{
let pricePerItem = document.getElementById("pricePerItem");

if(pricePerItem==null)
    return;
        
if(qty == 1)
    pricePerItem.innerHTML = "";
else
    pricePerItem.innerHTML = "Each $ "+currentProduct.offer_price;
}

function addKeyValue(obj, key, data){
obj[key] = data;
}

DetailsSection.propTypes = {
    currentProduct: PropTypes.object,
    dispatchAddToCart: PropTypes.func
}

const mapStateToProps = productDetail =>{
    return productDetail;
}

const mapDispatchToProps = (dispatch) =>{
    return {
        dispatchAddToCart: product => dispatch(addToCart(product))
    }
}

export default connect(mapStateToProps,mapDispatchToProps)(DetailsSection);