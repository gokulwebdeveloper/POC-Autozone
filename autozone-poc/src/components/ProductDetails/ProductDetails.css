.owl-prev {
    position: absolute;
    left: 0;
    left: 30px;
    height: 30px;
    text-align: center;
    transform: translateY(-50%);
    top: 50%;
    background: url(../../assets/img/core-img/long-arrow-left.svg) no-repeat !important;
    width: 30px;
    opacity: 0.4;
    border: none !important;
  }
  
  .owl-nav span {
    visibility: hidden;
  }
  
  button.owl-prev:focus {
    border:transparent 0px
  }
  
  .owl-next {
    position: absolute;
    right: 0;
    right: 30px;
    height: 30px;
    text-align: center;
    transform: translateY(-50%);
    top: 50%;
    background: url(../../assets/img/core-img/long-arrow-right.svg) no-repeat !important;
    width: 30px;
    opacity: 0.4;
  }
    
  button.owl-prev:focus, 
  button.owl-next:focus {
    outline: none;
  }