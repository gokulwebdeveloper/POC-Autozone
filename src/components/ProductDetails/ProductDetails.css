
  .form-control.quantitybox {
    width: 40px;
    height: 35px;
    margin-right: 5px;
    margin-left: 5px;
    text-align: center;
    font-weight: 600;
    font-size: 18px;
  }

  .single_product_desc .quantitybutton{
    min-width: 30px;
    padding: 0px;
    background-color: transparent;
    border: none;
    outline: none;
  }

  .single_product_desc .quantitybutton:focus{
    background-color: transparent;
    border: none;
    outline: none;
  }

  .single_product_details_area .single_product_desc .product-price .new-price{
    display: inline-block;
    font-family: "Ubuntu", sans-serif;
    font-size: 24px;
    color: #dc0345;
    font-weight: 700;
    text-decoration: none;
    margin-right: 0px;
  }

  .single_product_desc .priceperitem{
  font-size: 10px;
  height: 10px;
  margin: 10px;
  letter-spacing: normal;
  text-transform: none;
  font-weight: normal;
  color: #484848;
  }

  .single_product_desc .outofstock{
    color: #dc0345;
    height: 10px;
  }

.error-template {padding: 40px 15px;text-align: center;}
.error-actions {margin-top:15px;margin-bottom:15px;}
.error-actions .btn { margin-right:10px; }

.single_product_desc .rating-row{
    display: inline-flex;
    margin-bottom: -50px;
}

.single_product_desc .rating-row.warranty{
  color: #484848;
  margin-right: 10px;
  margin-left: 10px;
}

.single_product_desc .rating-row .rating-number{
  color: #484848;
  margin-right: 10px;
  margin-left: 10px;
}

.orderByMessage{
  color: #484848;
  margin-bottom: 10px;
}

.single_product_thumb .slick-slider button.slick-arrow.slick-prev,
.single_product_thumb .slick-slider button.slick-arrow.slick-next{
  margin-left: 10px;
  margin-right: 10px;
  z-index: 100;
  background-color: lightgray;
}

.single_product_thumb .slick-slider button.slick-arrow.slick-prev:hover, 
.single_product_thumb .slick-slider button.slick-arrow.slick-prev:focus,
.single_product_thumb .slick-slider button.slick-arrow.slick-next:hover, 
.single_product_thumb .slick-slider button.slick-arrow.slick-next:focus {
  outline: none;
  background-color: lightgray;
}

.single_product_details_area .single_product_thumb, .single_product_details_area .single_product_desc{
  padding: 10px;
}

.single_product_details_area {
  margin-left: 10px;
  margin-right: 10px;
}

.single_product_details_area .single_product_thumb .slick-list{
  margin-bottom: 15px;
}