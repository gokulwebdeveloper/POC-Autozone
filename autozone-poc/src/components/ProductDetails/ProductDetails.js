import React from "react";
import '../../requiredLoader'
import 'owl.carousel/dist/assets/owl.carousel.css';
import OwlCarousel from 'react-owl-carousel';
import '../ProductDetails/ProductDetails.css';
import Button from '../../utlis/button/Button'

const ProductDetails = () => {
    var styleDisplayNone = {
        display:'none',
     };

     var styleowlstage = {
        transform: 'translate3d(-2295px, 0px, 0px)',
         transition: 'all 1s ease 0s',
         width: 3213
    };

    var styleowlitems = {
       width: 459,
    };

    function changeFavStatus() {
        var currentCntrl = document.getElementById("favme");

        if(currentCntrl==null)
            return;

        if(!currentCntrl.className.includes("active"))
            currentCntrl.className = "favme fa fa-heart active";
        else
            currentCntrl.className = "favme fa fa-heart";
    }

    return (
        <section className="single_product_details_area d-flex align-items-center">
            <div className="single_product_thumb clearfix">
            <OwlCarousel
     className={'owl-carousel'}
     loop={true}
     nav={true}
     dots={false}
     autoplay={true}
     autoplayTimeout={5000}
     autoplaySpeed={1000}
     fluidSpeed={1000}
     items={1}
     responsiveRefreshRate={200}
>

<div><img src="src/utlis/img/product-img/product-big-1.jpg" alt=""/></div>
<div><img src="src/utlis/img/product-img/product-big-2.jpg" alt=""/></div>
<div><img src="src/utlis/img/product-img/product-big-3.jpg" alt=""/></div>
</OwlCarousel>
</div>

        {/* <!-- Single Product Description --> */}
        <div className="single_product_desc clearfix">
            <span>mango</span>
            <a href="cart.html">
                <h2>One Shoulder Glitter Midi Dress</h2>
            </a>
            <p className="product-price"><span className="old-price">$65.00</span> $49.00</p>
            <p className="product-desc">Mauris viverra cursus ante laoreet eleifend. Donec vel fringilla ante. Aenean finibus velit id urna vehicula, nec maximus est sollicitudin.</p>

            {/* <!-- Form --> */}
            <form className="cart-form clearfix" method="post">

                {/* <!-- Select Box --> */}
                <div className="select-box d-flex mt-50 mb-30">
                    <select id="productSize" className="nice-select mr-5">
                        <option >Size: XL</option>
                        <option >Size: X</option>
                        <option >Size: M</option>
                        <option >Size: S</option>
                    </select>
                    <select id="productColor" className="nice-select mr-5">
                        <option >Color: Black</option>
                        <option >Color: White</option>
                        <option >Color: Red</option>
                        <option >Color: Purple</option>
                    </select>
                </div>
                {/* <!-- Cart & Favourite Box --> */}
                <div className="cart-fav-box d-flex align-items-center">
                    <Button btnText="Add to Cart"/>
                    {/* <!-- Favourite --> */}
                    <div className="product-favourite ml-4" onClick={changeFavStatus}>
                        <a id="favme" className="favme fa fa-heart"></a>
                    </div>
                </div>
        </form>
        </div>
        </section>
    );
};

export default ProductDetails;