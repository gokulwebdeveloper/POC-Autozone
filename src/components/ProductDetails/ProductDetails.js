import React from 'react';
import DetailsSection from './DetailsSection';
import ProductNotFound from './ProductNotFound';
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import {getProductDetail} from '../../Redux/actions';

const ProductDetails = (props) => {
    const prodId = props.match?props.match.params.id:null;

    if(prodId){
        if(props.productDetail)
            return <DetailsSection currentProduct={props.productDetail.data[0]} />
        else
            return <ProductNotFound/>
    }
    else
       return <ProductNotFound/>
};

ProductDetails.propTypes = {
    props: PropTypes.object,
}

const mapStateToProps = ({productDetail}) =>{
    return {productDetail};
}

const mapDispatchToProps = {getProductDetail}

export default connect(mapStateToProps, mapDispatchToProps)(ProductDetails);