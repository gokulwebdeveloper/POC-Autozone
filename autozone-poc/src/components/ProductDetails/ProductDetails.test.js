import React from "react";
import { unmountComponentAtNode } from "react-dom";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import '@testing-library/jest-dom/extend-expect';
import ProductDetails from "./ProductDetails";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("section");
  document.body.appendChild(container);
});

describe("renders component with images", () => {
  const details = shallow(<ProductDetails />);
  it("SnapShot Match for Product Details screen", () => {
    const page = renderer.create(<ProductDetails />).toJSON();
    expect(page).toMatchSnapshot();
  });
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});