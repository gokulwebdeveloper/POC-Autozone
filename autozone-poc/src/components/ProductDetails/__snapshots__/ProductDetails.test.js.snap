// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component with images SnapShot Match for Product Details screen 1`] = `
<section
  className="single_product_details_area d-flex align-items-center"
>
  <div
    className="single_product_thumb clearfix"
  >
    <div
      className="owl-carousel owl-carousel"
    >
      <div>
        <img
          alt=""
          src="src/utlis/img/product-img/product-big-1.jpg"
        />
      </div>
      <div>
        <img
          alt=""
          src="src/utlis/img/product-img/product-big-2.jpg"
        />
      </div>
      <div>
        <img
          alt=""
          src="src/utlis/img/product-img/product-big-3.jpg"
        />
      </div>
    </div>
  </div>
  <div
    className="single_product_desc clearfix"
  >
    <span>
      mango
    </span>
    <a
      href="cart.html"
    >
      <h2>
        One Shoulder Glitter Midi Dress
      </h2>
    </a>
    <p
      className="product-price"
    >
      <span
        className="old-price"
      >
        $65.00
      </span>
       $49.00
    </p>
    <p
      className="product-desc"
    >
      Mauris viverra cursus ante laoreet eleifend. Donec vel fringilla ante. Aenean finibus velit id urna vehicula, nec maximus est sollicitudin.
    </p>
    <form
      className="cart-form clearfix"
      method="post"
    >
      <div
        className="select-box d-flex mt-50 mb-30"
      >
        <select
          className="nice-select mr-5"
          id="productSize"
        >
          <option>
            Size: XL
          </option>
          <option>
            Size: X
          </option>
          <option>
            Size: M
          </option>
          <option>
            Size: S
          </option>
        </select>
        <select
          className="nice-select mr-5"
          id="productColor"
        >
          <option>
            Color: Black
          </option>
          <option>
            Color: White
          </option>
          <option>
            Color: Red
          </option>
          <option>
            Color: Purple
          </option>
        </select>
      </div>
      <div
        className="cart-fav-box d-flex align-items-center"
      >
        <button
          className="btn essence-btn"
        >
          Add to Cart
        </button>
        <div
          className="product-favourite ml-4"
          onClick={[Function]}
        >
          <a
            className="favme fa fa-heart"
            id="favme"
          />
        </div>
      </div>
    </form>
  </div>
</section>
`;
